nixos/modules/services/networking/nomad.nix

```diff
@@ -0,0 +1,129 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.nomad;
+  dataDir = "/var/lib/nomad";
+  format = pkgs.formats.json { };
+in
+{
+  ##### interface
+  options = {
+    services.nomad = {
+      enable = mkEnableOption "A Distributed, Highly Available, Datacenter-Aware Scheduler";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.nomad;
+        defaultText = "pkgs.nomad";
+        description = ''
+          The package used for the Nomad agent and CLI.
+        '';
+      };
+
+      dropPrivileges = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether the nomad agent should be run as a non-root nomad user.
+        '';
+      };
+
+      interface = {
+        advertise = {
+          http = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = "The name of the interface to advertise http to.";
+          };
+          rpc = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = "The name of the interface to advertise rpc to.";
+          };
+          serf = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = "The name of the interface to advertise serf to.";
+          };
+        };
+      };
+
+      settings = mkOption {
+        type = format.type;
+        default = { };
+        description = ''
+          Configuration for Nomad. See the <link xlink:href="https://www.nomadproject.io/docs/configuration">documentation</link>
+          for supported values.
+        '';
+      };
+    };
+  };
+
+  ##### implementation
+  config = mkIf cfg.enable {
+    users = {
+      groups.nomad.gid = config.ids.gids.nomad;
+      users.users.nomad = {
+        uid = config.ids.uids.nomad;
+        group = "nomad";
+      };
+    };
+
+    environment = {
+      environment.etc."nomad.json".source = format.generate "nomad.json" cfg.settings;
+      systemPackages = [ cfg.package ];
+    };
+
+    systemd.tmpfiles.rules = [
+      "d '${dataDir}' - nomad nomad - -"
+    ];
+
+    systemd.services.nomad = {
+      description = "A Distributed, Highly Available, Datacenter-Aware Scheduler";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      restartTriggers = [ "/etc/nomad.json" "/etc/nomad-addrs.json" ];
+
+      serviceConfig = {
+        ExecStart = "${cfg.package}/bin/nomad agent -config=/etc/nomad.json -config=/etc/nomad-addrs.json";
+        User = optionalString cfg.dropPrivileges "nomad";
+        Restart = "on-failure";
+      };
+
+      path = with pkgs; [ iproute gnugrep gawk ] ++ cfg.package;
+      preStart = ''
```

Comment:
This works? How are permutations granted in the case this doesn't run as `root`?

Reply:
I think this works because it's not trying to modify the IP configuration, rather it reads it. Reading works as non-root too.

Reply:
I guess I should have clarified. I meant the line that writes to `/etc/nomad-addrs.json`.